GH2101: Disable report printing if verbosity of logger is Quiet

deniszykov opened the pull request:
fix for https://github.com/cake-build/cake/issues/2101

deniszykov commented:
@microsoft-github-policy-service agree

augustoproiete commented on src/Cake.Core/CakeReportPrinter.cs:
```diff
@@ -30,13 +30,18 @@ public CakeReportPrinter(IConsole console, ICakeContext context)
+            {
+                return;
+            }
+
```
Thanks for the PR @deniszykov! As you can see in [this comment in the original issue](https://github.com/cake-build/cake/issues/2101#issuecomment-376160140) we wouldn't want to change the semantics of the `Quiet` verbosity in order to hide the report - this means users don't have a way to suppress the log messages and see the report anymore.

Could you add a new command-line argument to Cake called `--no-report` that hides the report? (regardless of the Verbosity)

gep13 replied:
I have taken the liberty to update this PR with the suggested changes.

gep13 committed:
(#2101) Allow report summary to not be shown

As part of a normal Cake execution, a report summary is always shown at
the end. This includes information about what tasks were ran, how long
they took, etc.  However, there are those that don't want to allow this
to happen.  This commit introduces a new configuration option, which
allows the report summary to not be shown. This can be done in the
usual way with a command line parameter, cake.config file entry, or an
environment variable.

gep13 commented:
@deniszykov rather than create a new PR, I have taken the liberty to update this PR using the suggested approach discussed in the original issue.

Please let me know if you have any concerns about me doing this.

deniszykov commented:
> @deniszykov rather than create a new PR, I have taken the liberty to update this PR using the suggested approach discussed in the original issue.
> 
> Please let me know if you have any concerns about me doing this.

I don't mind, and thanks. I was actually thinking of adding another logging level below Quiet.  But I solved my problem differently and forgot about this PR.

gep13 commented:
@deniszykov said...
I don't mind, and thanks.

Perfect! I have started a process of reviewing some of the older PR's that we have, and I stumbled across your one.

gep13 commented on src/Cake.Cli/Hosts/BuildScriptHost.cs:
```diff
@@ -88,7 +96,8 @@ private async Task<CakeReport> internalRunTargetAsync()
                 var report = await Engine.RunTargetAsync(_context, _executionStrategy, Settings).ConfigureAwait(false);
 
-                if (report != null && !report.IsEmpty)
+                var noReportEnabled = _configuration.GetValue("Settings_NoReport") ?? bool.FalseString;
```
@devlead I tried to use the `Constants` class here, rather than the hard coded `Settings_NoReport`, but it is marked `internal`.  Should I create a new `Constants` class in the Cake.Cli project?

devlead replied:
@gep13 If it's just used in Cake.Cli it makes sense to add a new Constants class in Cake.Cli project.

devlead committed:
Add --no-report option to Frosting too

- Add --no-report flag to Cake.Frosting default commands
- Introduce Cake.Cli.Infrastructure.Constants for Settings_NoReport key
- Extend RemainingArgsParser.ToCakeArguments with noReport parameter
- BuildScriptHost uses constant when checking report display
- Minor cleanup: TryAdd/collection expressions in arg parsing

devlead reviewed (approved):
LGMT 👍

devlead commented:
@deniszykov & @gep13  your changes have been merged, thanks for your contribution 👍
